hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieDeltaStreamer.java

```diff
@@ -742,7 +742,7 @@ protected Pair<CompletableFuture, ExecutorService> startService() {
                   scheduledCompactionInstantAndRDD.isPresent() ? HoodieJavaRDD.of(scheduledCompactionInstantAndRDD.get().getRight()) : null);
               if (requestShutdownIfNeeded(lastWriteStatuses)) {
                 LOG.warn("Closing and shutting down ingestion service");
-                error = true;
+                waitAsyncServicesFinishAndStop();
```

Comment:
Shouldn't the shutdown in line 747 below take of shutting down async services too?
cc @nsivabalan